src/eventloop.jl

```diff
@@ -52,7 +57,63 @@ function waitloop()
     end
 end
 
-const idle_delay = Ref(0.5)
-function set_idle_delay(delay_secs::Float64=0.5)
-    idle_delay[] = delay_secs
+#=
+    Hold Channel Interface
+    This allows user processes to register and deregister
+    channels that allow user code to hold IJulia from sending
+    the idle status message until signaled.  This would have
+    been prettier if there were level-triggered events in
+    Julia base, but this will do okay for now.
```

Comment:
It's not clear to me why we're using `Channel` and not `Condition` here.   If we used `Condition`, then tasks could just call `notify()` when they are idle, no?